### PR TITLE
Upgrade CI to Go 1.17

### PR DESCRIPTION
**Pull request by sergei-ivanov**
Signed-off-by: Sergei Ivanov <sergei_ivanov@mail.ru>

### CODE REVIEW

**Comment by sergei-ivanov** [comment]
Forgot to update CI environment in #11 to line up with Dockerfile

File: .goreleaser.yml [hunk 1]
```diff
@@ -22,6 +22,12 @@ builds:
   ignore:
     - goos: darwin
       goarch: '386'
+    - goos: freebsd
```
Comment (invidian) [inline .goreleaser.yml:30]: Hmm, I don't see those in https://github.com/hashicorp/terraform-provider-scaffolding/blob/6908b6b7ada594861f7db42f3d47ad37a221e416/.goreleaser.yml and this file is directly pulled from there.

Is this something we could report/contribute? Why is this needed in general?

**Review by invidian — COMMENTED** [review]
LGTM, just one question.

File: .goreleaser.yml (hunk 1 above)
Comment (sergei-ivanov) [reply .goreleaser.yml:30]: I agree, I looked at the same reference project in the last couple of days and noticed that it only excludes macOS i386 flavour.

To be honest, I don't know at which stage the filtering happens, but I can only see a subset of flavours in terraform registry. I checked terraform itself and AWS provider. 

The only reason I made this change in the config is because goreleaser CI task takes ages and I wanted to speed it up a bit.

Comment (invidian) [reply .goreleaser.yml:30]: Make sense. It seems here are recommended builds documented: https://www.terraform.io/registry/providers/os-arch

That gives the following matrix:
Selection_529

I'm fine with updating the config according to the matrix.

Do you mind creating an issue about it in https://github.com/hashicorp/terraform-provider-scaffolding? I guess it would be nice if the documentation and configs could be aligned.

Comment (sergei-ivanov) [reply .goreleaser.yml:30]: Oh, great find. I haven't come across that page before. I'll raise a PR upstream in the scaffolding project. I did suspect that Windows ARM flavours were not needed either.

Comment (invidian) [reply .goreleaser.yml:30]: I wonder if we should also consider platforms mentioned https://releases.hashicorp.com/terraform/1.1.4/. All that doesn't seem very consistent.

Comment (sergei-ivanov) [reply .goreleaser.yml:30]: That is exactly what I based my original exclusions on. I looked at the same page yesterday.

Comment (invidian) [reply .goreleaser.yml:30]: Let's leave FreeBSD arm build then, others can be ignored :+1: And please put links to https://releases.hashicorp.com/terraform/1.1.4/ and https://www.terraform.io/registry/providers/os-arch in a comment, so we know from where this all comes from.

And thanks for patience, contribution, thoroughness and understanding, I highly appreciate it!

Comment (sergei-ivanov) [reply .goreleaser.yml:30]: OK, to summarise:

1. I shall keep the goreleaser config as it is on the PR now
2. I shall add clarifying comments to the list of excluded flavours
3. I shall raise an issue or PR upstream in the scaffolding project to update the reference implementation

I'll do 2. a bit later in the afternoon (UK time zone).

Comment (invidian) [reply .goreleaser.yml:30]: We can also ignore darwin 386 I think? Otherwise sounds good.

Comment (sergei-ivanov) [reply .goreleaser.yml:30]: Done, sorry that it took more than a day to add a couple of lines :)
I have raised https://github.com/hashicorp/terraform-provider-scaffolding/issues/91 upstream as well.

**Review by invidian — APPROVED** [review]
LGTM, thank you!

**Comment by sergei-ivanov** [comment]
Splendid. Let's try to create a new tag and release it. Also, can you please either send your public GPG key to the keyservers, or add it to this repo, or register with Keybase?
```
$ gpg --recv-keys 97187FA1271EC224
gpg: keyserver receive failed: No data
```

**Comment by invidian** [comment]
I need to add changelog still, but sure.

My key is there:
```console
$ gpg --keyserver keyserver.ubuntu.com --recv-keys 97187FA1271EC224
gpg: key 97187FA1271EC224: "Mateusz Gozdek <mgozdekof@gmail.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```

I don't know what keyserver you use :) 

You can also do `curl -L https://github.com/invidian.gpg | gpg --import`.

**Comment by sergei-ivanov** [comment]
Ah yes, everything works with explicitly using ubuntu keyserver, thanks! And TIL about GitHub having a public URL for downloading committers' public keys -- amazing!

**Comment by invidian** [comment]
Note, that I sign commits with `85C07FB75C70CF6A`, but Terraform Registry does not support ed25519 keys yet, so I have to use my old key `97187FA1271EC224` for release artifacts signing.

**Comment by sergei-ivanov** [comment]
I see the new release has already filtered through to terraform registry, and we are going to test it with my colleagues shortly. Thanks a lot for the effort!
